internal/cmd/egctl/config_ratelimit.go

```diff
@@ -0,0 +1,206 @@
+	defaultConfigMapKey       = "envoy-gateway.yaml"   // TODO: make this configurable until EG support
+)
+
+func ratelimitConfigCommand() *cobra.Command {
+
+	var (
+		namespace string
+	)
+
+	rlConfigCmd := &cobra.Command{
+		Use:     "envoy-ratelimit",
+		Aliases: []string{"rl"},
+		Long:    `Retrieve the relevant rate limit configuration from the Rate Limit instance`,
+		Example: `  # Retrieve rate limit configuration
+  egctl config ratelimit
+
+  # Retrieve rate limit configuration with short syntax
+  egctl c rl
+`,
+		Run: func(c *cobra.Command, args []string) {
+			cmdutil.CheckErr(runRateLimitConfig(c, namespace))
+		},
+	}
+
+	rlConfigCmd.Flags().StringVarP(&namespace, "namespace", "n", defaultRateLimitNamespace, "Specific a namespace to get resources")
+	return rlConfigCmd
+}
+
+func runRateLimitConfig(c *cobra.Command, ns string) error {
+
+	cli, err := getCLIClient()
+	if err != nil {
+		return err
+	}
+
+	out, err := retrieveRateLimitConfig(cli, ns)
+	if err != nil {
+		return err
+	}
+
+	_, err = fmt.Fprintln(c.OutOrStdout(), string(out))
+	return err
+}
+
+func retrieveRateLimitConfig(cli kubernetes.CLIClient, ns string) ([]byte, error) {
+
+	// Before retrieving the rate limit configuration
+	// we make sure that the global rate limit feature is enabled
+	if enable, err := checkEnableGlobalRateLimit(cli); !enable {
+		return nil, fmt.Errorf("global rate limit feature is not enabled")
+	} else if err != nil {
+		return nil, fmt.Errorf("failed to get global rate limit status: %w", err)
+	}
+
+	// Filter out all rate limit pods in the Running state
+	rlNN, err := fetchRunningRateLimitPods(cli, ns, ratelimit.RateLimitLabelSelector())
+	if err != nil {
+		return nil, err
+	}
+
+	// In fact, the configuration of multiple rate limit replicas are the same.
+	// After we filter out the rate limit Pods in the Running state,
+	// we can directly use the first pod.
+	rlPod := rlNN[0]
+	fw, err := portForwarder(cli, rlPod, rateLimitDebugPort)
+	if err != nil {
+		return nil, fmt.Errorf("failed to initialize pod-forwarding for %s/%s: %w", rlPod.Namespace, rlPod.Name, err)
+	}
+
+	return extractRateLimitConfig(fw, rlPod)
+}
+
+// fetchRunningRateLimitPods gets the rate limit Pods, based on the labelSelectors.
+// It further filters out only those rate limit Pods that are in "Running" state.
+func fetchRunningRateLimitPods(cli kubernetes.CLIClient, namespace string, labelSelector []string) ([]types.NamespacedName, error) {
+
+	// Since multiple replicas of the rate limit are configured to be equal,
+	// we do not need to use the pod name to obtain the specified pod.
+	rlPods, err := cli.PodsForSelector(namespace, labelSelector...)
+	if err != nil {
+		return nil, err
+	}
+
+	rlNN := []types.NamespacedName{}
+	for _, rlPod := range rlPods.Items {
+		rlPodNsName := types.NamespacedName{
+			Namespace: rlPod.Namespace,
+			Name:      rlPod.Name,
+		}
+		if rlPod.Status.Phase != "Running" {
```

Comment:
> we should check the whether `PodConditions` contain `corev1.ContainersReady` instead of checking pod is running. the RL pod is Running but not Ready it means that the Readiness probe is failing, so it still does not receive any traffic, including the request egctl about to send.

I think we should use `Ready`, which is better than `ContainersReady`. ContainersReady only means that the container is ready, but does not ensure that the Pod can accept traffic at this time. @shawnh2

Reply:
right, SGTM